### PR TITLE
Feat/ROE-2375: Resume journey with `transactionId` and `submissionId` in URL

### PR DESCRIPTION
**Pull request by mwejuli-ch**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2375

### Change description

Adds `transactionId` and `submissionId` in URL for the resume journey

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### CODE REVIEW

File: test/controllers/resume.submission.controller.spec.ts [hunk 1]
```diff
@@ -185,6 +186,38 @@ describe("Resume submission controller", () => {
...
+      OVERSEAS_ENTITY_ID,
+      { headers: PAYMENT_HEADER },
+      baseURL
+    );
+    expect(mockInfoRequest).toHaveBeenCalledWith( expect.anything(), `Payments Session created on Resume link with, ${errorMsg}`);
+    expect(mockGetOverseasEntity).toHaveBeenCalledTimes(1);
+    expect(mockSetExtraData).toHaveBeenCalledTimes(1);
+    expect(mockCreateAndLogErrorRequest).not.toHaveBeenCalled();
+    expect(mockMapTrustApiReturnModelToWebModel).toHaveBeenCalledTimes(1);
+    expect(mockIsActiveFeature).toHaveBeenCalledTimes(2);
```
Comment (dlloyd1-ch) [inline test/controllers/resume.submission.controller.spec.ts:220]: Can we add an assertion that checks that `getUrlWithTransactionIdAndSubmissionId()` has been called when the flag is set to `true`?

Comment (mwejuli-ch) [reply test/controllers/resume.submission.controller.spec.ts:225]: done!

**Comment by markpit** [comment]
LGTM, will await dan's comment response

**Review by dlloyd1-ch — APPROVED** [review]

**Review by mwestacott — COMMENTED** [review]
I can't see a code change for the Pay Now link behaviour described in the story. Is one needed or not?

**Comment by mwejuli-ch** [comment]
> I can't see a code change for the Pay Now link behaviour described in the story. Is one needed or not?

I do not think we need one coz we're only doing resume (from the `Your filings` tab). But I stand to be corrected..

**Comment by mwestacott** [comment]
> > I can't see a code change for the Pay Now link behaviour described in the story. Is one needed or not?
> 
> I do not think we need one coz we're only doing resume (from the `Your filings` tab). But I stand to be corrected..

If you've tested the scenario and it's working you're probably fine. The transaction id appears on the confirmation page as the reference number and is read from the session but it's probably not needed within the URL. When we change `getApplicationData` to read from the API not session it will hopefully all magically just work.

**Review by mwestacott — APPROVED** [review]

**Review by lduranteau — APPROVED** [review]

**Pull request merged**